client.go

```diff
@@ -1112,7 +1119,7 @@ func (c *Client[TTx]) ID() string {
 	return c.config.ID
 }
 
-func insertParamsFromConfigArgsAndOptions(config *Config, args JobArgs, insertOpts *InsertOpts) (*riverdriver.JobInsertFastParams, *dbunique.UniqueOpts, error) {
+func insertParamsFromConfigArgsAndOptions(archetype *baseservice.Archetype, config *Config, args JobArgs, insertOpts *InsertOpts) (*riverdriver.JobInsertFastParams, *dbunique.UniqueOpts, error) {
```

Comment:
Now that we're passing in a couple of fields from `Client` to this (neither of which was needed 1mo ago), it might be cleaner and less confusing to make it a method on `Client` instead. Borderline 🤷‍♂️

Reply:
Yeah true.

I took a shot at it, but in the end, decided to leave as is.

The trouble is that `insertParamsFromConfigArgsAndOptions` gets called from a few other non-client components that make some annoying changes required:

* `periodic_job.go` needs it in `PeriodicJobBundle` so it can insert periodic jobs. We could embed a client on `PeriodicJobBundle` instead of a client config, but then `PeriodicJobBundle` needs to take a generic parameter so it can put a type on `Client[TTx]`, which is super annoying.
* `producer_test.go` uses it. It's workable just to create a client there and call the function on that, but sort of annoying to add the extra client that's not really used for anything else.

Let's keep an eye on this one. It's definitely a refactor that's possible, and we should do it if things get anymore gnarly, but this might be the last addition it needs in a while, so maybe okay for now.